tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/polyparser/ParsePool.scala

```diff
@@ -17,6 +18,12 @@ case class ParsePool(parses: Iterable[(PolytreeParse, Double)]) {
   def toNbestList: NbestList = {
     NbestList(parses)
   }
+
+  @transient lazy val indexedParses = parses.toIndexedSeq
+
+  def chooseRandomParse: PolytreeParse = {
+    indexedParses(Random.nextInt(indexedParses.size))._1
```

Comment:
No, but it's during training, not during actual parser execution, so repeatability isn't super important here.